lib/dry/types/compiler.rb

```diff
@@ -39,45 +40,19 @@ def visit_array(node)
         registry['array'].member(call(node))
       end
 
-      def visit_form_array(node)
-        registry['form.array'].member(call(node))
-      end
-
-      def visit_json_array(node)
-        registry['json.array'].member(call(node))
-      end
-
       def visit_hash(node)
         constructor, schema = node
         merge_with('hash', constructor, schema)
       end
 
-      def visit_form_hash(node)
-        if node
-          constructor, schema = node
-          merge_with('form.hash', constructor, schema)
-        else
-          registry['form.hash']
-        end
-      end
-
-      def visit_json_hash(node)
-        if node
-          constructor, schema = node
-          merge_with('json.hash', constructor, schema)
-        else
-          registry['json.hash']
-        end
-      end
-
-      def visit_key(node)
+      def visit_member(node)
         name, types = node
         { name => visit(types) }
       end
 
       def merge_with(hash_id, constructor, schema)
         registry[hash_id].__send__(
-          constructor, schema.map { |key| visit(key) }.reduce({}, :merge)
+          constructor, schema.first.map { |key| visit(key) }.reduce({}, :merge)
```

Comment:
Right, but the hash visitor makes sense as it is now, so let's leave it like that